### PR TITLE
feat(List): filter dock mode

### PR DESCRIPTION
**Pull request by jsomsanith**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
The filter in toolbar is an input

**What is the new behavior?**
The filter is a button that reveal the input on click
The full spec is [here](http://guidelines.talend.com/document/92132#/ui-controls/find-box-filter-br)

**Does this PR introduce a breaking change?**

- [x] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...
Fixed onFilter() callback arguments from (value, event) to (event, value)

**Other information**:
Change the version in BREAKIN_CHANGED_LOG.md before merge

### CODE REVIEW

File: packages/components/src/List/Toolbar/Filter/Filter.component.js [hunk 1]
```diff
@@ -1,68 +1,161 @@
...
+	case ENTER_KEY:
+		if (enterAction) {
+			enterAction(event);
+		}
+		break;
+	default:
+		break;
+	}
+}
+function FilterInput(props) {
```
Comment (jmfrancois) [inline packages/components/src/List/Toolbar/Filter/Filter.component.js]: export and add unit tests on it

**Comment by jmfrancois** [comment]
two screenshots are white. may be those views are broken with your changes.

File: packages/components/src/List/Toolbar/Filter/Filter.component.js (hunk 1 above)
Comment (jsomsanith) [inline packages/components/src/List/Toolbar/Filter/Filter.component.js]: This FilterInput is tested (snapshot and event behavior) in Filter tests

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jsomsanith** [comment]
@jmfrancois `two screenshots are white. may be those views are broken with your changes.`--> fixed, it was because of the stories rename, I updated the screenshot config.

File: packages/components/src/List/Toolbar/Filter/Filter.component.js [hunk 2]
```diff
@@ -1,68 +1,161 @@
...
+import classNames from 'classnames';
 import DebounceInput from 'react-debounce-input';
 import FormControl from 'react-bootstrap/lib/FormControl';
-import Button from 'react-bootstrap/lib/Button';
+import { Action } from '../../../Actions';
 import Icon from '../../../Icon';
+import theme from './Filter.scss';
+
+const ESC_KEY = 27;
+const ENTER_KEY = 13;
```
Comment (talend-glorieux) [inline packages/components/src/List/Toolbar/Filter/Filter.component.js]: I'm starting to see those kind of key handling constants more and more. 
Do you think we should extract them into a shared library?

File: packages/components/src/List/Toolbar/Filter/Filter.component.js [hunk 3]
```diff
@@ -1,68 +1,161 @@
...
+				icon="talend-search"
+				bsStyle="link"
+			/>
+		);
 	}
 
 	function onSubmit(event) {
 		event.preventDefault();
-		return onFilter(this.value, event);
+		return onFilter(event, event.target.search.value);
```
Comment (talend-glorieux) [inline packages/components/src/List/Toolbar/Filter/Filter.component.js]: I think using `_.get` would be a good thing here. You can't be sure the entire `event.target.search.value` exists.

File: packages/components/src/List/Toolbar/Filter/Filter.component.js (hunk 2 above)
Comment (jsomsanith) [inline packages/components/src/List/Toolbar/Filter/Filter.component.js]: Would this extracted in a constant file in utils/ would be sufficient ?

File: packages/components/src/List/Toolbar/Filter/Filter.component.js (hunk 3 above)
Comment (jsomsanith) [inline packages/components/src/List/Toolbar/Filter/Filter.component.js]: Done

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

File: packages/components/src/List/Toolbar/Filter/Filter.component.js (hunk 2 above)
Comment (jsomsanith) [inline packages/components/src/List/Toolbar/Filter/Filter.component.js]: Extracted into an utils file

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://74.talend.surge.sh)

**Pull request merged**